src/app/page.tsx

```diff
@@ -1,116 +1,156 @@
-		chain,
-		client,
-	});
-
-	const currencyMetadata = useReadContract(getCurrencyMetadata, {
-		contract: currencyContract,
-		queryOptions: { enabled: !!currency },
-	});
-
-	const currencySymbol = currencyMetadata.data?.symbol || "";
-
-	const pricePerToken =
-		currencyMetadata.data && priceInWei !== null && priceInWei !== undefined
-			? Number(toTokens(priceInWei, currencyMetadata.data.decimals))
-			: null;
-
-	return (
-		<NftMint
-			contract={contract}
-			displayName={displayName || ""}
-			contractImage={contractMetadataQuery.data?.image || ""}
-			description={description || ""}
-			currencySymbol={currencySymbol}
-			pricePerToken={pricePerToken}
-			isERC1155={!!isERC1155Query.data}
-			isERC721={!!isERC721Query.data}
-			tokenId={tokenId}
-		/>
-	);
+import type { NFT } from "thirdweb";
+
+const chain = defineChain(defaultChainId);
+const contract = getContract({ address: defaultNftContractAddress, chain, client });
+const tokenId = defaultTokenId;
+
+// Define the possible ERC token types
+type ERCType = "ERC1155" | "ERC721" | "ERC20" | null;
+
+type CurrencyMetadata = {
+  decimals: number;
+  name: string;
+  symbol: string;
+};
+
+type ContractMetadata = {
+  name?: string; // Directly available in ERC20
+  description?: string;
+  image?: string;
+  data?: {
+    name?: string; // Nested inside "data" for ERC721 & ERC1155
+    description?: string;
+    image?: string;
+  };
+};
+
+type ClaimCondition = {
+  allowlistMerkleRoot?: `0x${string}`;
+  availableSupply?: bigint;
+  currency: string;
+  maxMintPerWallet?: bigint;
+  pricePerToken?: bigint;
+  pricePerUnit?: bigint;
+  startTimestamp: bigint;
+};
+
+type NFTInfo = {
+  displayName?: string;
+  description?: string;
+  pricePerToken?: number | null;
+  currencyMetadata?: CurrencyMetadata;
+  nft?: NFT;
+  metadata?: ContractMetadata;
+  claimCondition?: ClaimCondition;
+};
+
+type FetchResult = {
+  ercType: ERCType;
+  info: NFTInfo | null;
+  error?: string;
+};
+
+// Function to fetch NFT details, including ERC type and metadata
+async function fetchNFTInfo(): Promise<FetchResult> {
+  try {
+    // Determine ERC type by checking contract compatibility
+    const [isErc721, isErc1155] = await Promise.all([
+      isERC721({ contract }).catch(() => false),
+      isERC1155({ contract }).catch(() => false),
+    ]);
+
+    const ercType: ERCType = isErc1155 ? "ERC1155" : isErc721 ? "ERC721" : "ERC20";
+
+    // Define contract-specific logic in a lookup object
+    const claimConditionFetchers = {
+      ERC20: () => getActiveClaimCondition20({ contract }),
+      ERC721: () => getActiveClaimCondition721({ contract }),
+      ERC1155: () => getActiveClaimCondition1155({ contract, tokenId }),
+    };
+
+    // Fetch metadata and claim conditions in parallel
+    const [metadata, claimCondition]: [ContractMetadata | undefined, ClaimCondition | undefined] =
+      await Promise.all([
+        getContractMetadata({ contract }).catch(() => undefined),
```

Comment:
so you dont fetch the 1155 nft anymore? that's a problem because that means it will always show the same name/image/description even if its its a different 1155 you're buying!

Reply:
I really think something like 

```
const ercType = getERCType(...)
switch (ercType) {
  case "ERC20":
    return getERC20Info(...);
  case "ERC721":
    return getERC721Info(...);
  case "ERC1155":
    return getERC1155Info(...);
}
```

with each sub getERC... function in a *separate file* would be the cleanest in terms of readability. Yes there might be a little bit of duplication but worth the readability imo